lib/vector-core/src/fanout.rs

```diff
@@ -42,13 +37,66 @@ impl Fanout {
 
         let fanout = Self {
             sinks: vec![],
-            i: 0,
             control_channel: control_rx,
         };
 
         (fanout, control_tx)
     }
 
+    // TODO make a real error
```

Comment:
I did notice that all of these are `Box<dyn Sink<Event, Error = ()>`, so I'm not sure how real of an error we can make, but should probably resolve the TODO one way or another.

Reply:
Shoot, completely forgot about this comment.